src/module.c

```diff
@@ -10643,18 +10635,16 @@ RedisModuleCommandFilter *RM_RegisterCommandFilter(RedisModuleCtx *ctx, RedisMod
 /* Unregister a command filter.
  */
 int RM_UnregisterCommandFilter(RedisModuleCtx *ctx, RedisModuleCommandFilter *filter) {
-    listNode *ln;
+    int result;
 
     /* A module can only remove its own filters */
     if (filter->module != ctx->module) return REDISMODULE_ERR;
 
-    ln = listSearchKey(moduleCommandFilters,filter);
-    if (!ln) return REDISMODULE_ERR;
-    listDelNode(moduleCommandFilters,ln);
+    result = listSearchDel(moduleCommandFilters,filter);
+    if (!result) return REDISMODULE_ERR;
```

Comment:
Do we need the result variable? Why not simply

```suggestion
    if (!listSearchDel(moduleCommandFilters,filter)) return REDISMODULE_ERR;
```

Reply:
The reason was originally that the second check became long with the comment that dangled at the end. I don't really understand if the comment is even needed though, it maybe should just be a server assert.